src/vs/workbench/services/textmodelResolver/common/textModelResolverService.ts

```diff
@@ -174,13 +175,17 @@ export class TextModelResolverService extends Disposable implements ITextModelSe
 		@IInstantiationService private readonly instantiationService: IInstantiationService,
 		@IFileService private readonly fileService: IFileService,
 		@IUndoRedoService private readonly undoRedoService: IUndoRedoService,
-		@IModelService private readonly modelService: IModelService
+		@IModelService private readonly modelService: IModelService,
+		@IUriIdentityService private readonly uriIdentityService: IUriIdentityService,
 	) {
 		super();
 		this._register(new ModelUndoRedoParticipant(this.modelService, this, this.undoRedoService));
 	}
 
 	async createModelReference(resource: URI): Promise<IReference<IResolvedTextEditorModel>> {
+
+		resource = this.uriIdentityService.asCanonicalUri(resource);
```

Comment:
@bpasero this is the brute force way which might be OK... However, there are still all callers that we might need to visit and should maybe move this call to where it matter most, e.g where we actually branch off into the file service. E.g here: https://github.com/microsoft/vscode/blob/72a55628bc3253cfc2668d9d523830d188d1e3d3/src/vs/workbench/services/textmodelResolver/common/textModelResolverService.ts#L58

All other places shouldn't be affected by URI "normalization" because only file systems can contribute rules.